pkg/newlog/cmd/newlogd.go

```diff
@@ -0,0 +1,1281 @@
+	tmpStr2 := strings.SplitN(tmpStr1, ".log", 2)
+	return tmpStr2[0]
+}
+
+// at bootup, move the collected log files from previous life
+func findMovePrevLogFiles(movefile chan fileChanInfo) string {
+	files, err := ioutil.ReadDir(collectDir)
+	if err != nil {
+		log.Fatal("findMovePrevLogFiles: read dir ", err)
+	}
+
+	// get EVE version and partition, UUID may not be available yet
+	getEveInfo()
+
+	// remove any gzip file the name starts them 'Tempfile', it crashed before finished rename in dev/app dir
+	cleanGzipTempfiles(uploadDevDir)
+	cleanGzipTempfiles(uploadAppDir)
+
+	// on prev life's dev-log and app-log
+	for _, f := range files {
+		isDev := strings.HasPrefix(f.Name(), devPrefix)
+		isApp := strings.HasPrefix(f.Name(), appPrefix)
+		if !f.IsDir() && (isDev && len(f.Name()) > len(devPrefix) || isApp && len(f.Name()) > len(appPrefix)) {
+			var fileinfo fileChanInfo
+			prevLogFile := collectDir + "/" + f.Name()
+			fileinfo.tmpfile = prevLogFile
+			fileinfo.isApp = isApp
+			fileinfo.inputSize = int32(f.Size())
+
+			movefile <- fileinfo
+		}
+	}
+
+	return formatAndGetMeta("")
+}
+
+func trigMoveToGzip(fileinfo fileChanInfo, stats *statsLogFile, appUUID string, moveChan chan fileChanInfo, timerTrig bool) {
+	// check filesize over limit if not triggered by timeout
+	if !timerTrig && stats.size < maxLogFileSize {
+		return
+	}
+
+	if err := stats.file.Close(); err != nil {
+		log.Fatal(err)
+	}
+
+	isApp := appUUID != ""
+	fileinfo.isApp = isApp
+	fileinfo.inputSize = stats.size
+	fileinfo.tmpfile = stats.file.Name()
+
+	moveChan <- fileinfo
+
+	if isApp { // appM stats and logfile is created when needed
+		delete(appStatsMap, appUUID)
+		return
+	}
+
+	// reset stats data and create new logfile for device
+	stats.size = 0
+	stats.file = startTmpfile(collectDir, devPrefix)
+	stats.starttime = time.Now()
+
+	_, err := stats.file.WriteString(logmetaData + "\n") // write the metadata in the first line of logfile
+	if err != nil {
+		log.Fatal("trigMoveToGzip: write metadata line ", err)
+	}
+}
+
+func checkLogTimeExpire(fileinfo fileChanInfo, devStats *statsLogFile, moveChan chan fileChanInfo) {
+	// check device log file
+	if devStats.file != nil && int(time.Since(devStats.starttime).Seconds()) > logfileDelay {
+		trigMoveToGzip(fileinfo, devStats, "", moveChan, true)
+	}
+
+	// check app log files
+	for appuuid, appM := range appStatsMap {
+		if appM.file != nil && int(time.Since(appM.starttime).Seconds()) > logfileDelay {
+			trigMoveToGzip(fileinfo, &appM, appuuid, moveChan, true)
+		}
+	}
+}
+
+// for dev, returns the meta data, and for app, return the appName
+func formatAndGetMeta(appuuid string) string {
+	var appName string
+	if appuuid != "" {
+		for _, appD := range domainUUID { // cycle through the domainUUID map and find the UUID and appName
+			if appD.appUUID == appuuid {
+				appName = appD.appName
+				return appName
+			}
+		}
+	}
+	metaStr := logs.LogBundle{
```

Comment:
Ditto; json encode of protoc generated struct doesn't make much sense.

Reply:
@eriknordmark i tried to reuse the api defined for current log, just happens the json also was handled in the generated go files. otherwise, i need to define some new go api with json which i don't see anyone else in EVE/cloud is doing it

Reply:
Shouldn't the protoc internal fields appear in the json encoded output? If not, then why is it complaining about copying the lock?

Reply:
they are not in the final json representation string, due to the 'omitempty' definition. but i guess they have some intermediate steps may do copy.